src/c_wrapper.jl

```diff
@@ -29,43 +29,49 @@ function SCS_solve(T::Union{Type{Direct}, Type{Indirect}},
         m::Int, n::Int, A::SCSVecOrMatOrSparse, b::Array{Float64},
         c::Array{Float64}, f::Int, l::Int, q::Array{Int}, s::Array{Int},
         ep::Int, ed::Int, p::Array{Float64},
-        primal_sol::Vector{Float64}=Float64[],
-        dual_sol::Vector{Float64}=Float64[],
-        slack::Vector{Float64}=Float64[];
+        primal_sol::Vector{Float64}=zeros(n),
+        dual_sol::Vector{Float64}=zeros(m),
+        slack::Vector{Float64}=zeros(m);
         options...)
 
     n > 0 || throw(ArgumentError("The number of variables in SCSModel must be greater than 0"))
     m > 0 || throw(ArgumentError("The number of constraints in SCSModel must be greater than 0"))
 
-    managed_matrix = ManagedSCSMatrix(m, n, A)
-    matrix = Ref(SCSMatrix(managed_matrix))
-    settings = Ref(SCSSettings(T; options...))
-    data = Ref(SCSData(m, n, Base.unsafe_convert(Ptr{SCSMatrix}, matrix), pointer(b), pointer(c), Base.unsafe_convert(Ptr{SCSSettings},settings)))
-
-    cone = Ref(SCSCone(f, l, q, s, ep, ed, p))
-    info = Ref(SCSInfo())
-
     ws = (:warm_start=>true) in options
+    sizes_correct = length(primal_sol) == n && length(dual_sol) == length(slack) == m
 
-    if ws && length(primal_sol) == n && length(dual_sol) == m && length(slack) == m
-        x = primal_sol
-        y = dual_sol
-        s = slack
+    if sizes_correct
+        if !ws
+            primal_sol = fill!(primal_sol, 0.0)
```

Comment:
`fill!(primal_sol, 0.0)` (assignment isn't needed). Same for the other two.

Reply:
that's true, but only since these are `Vector`s (we require this in the signature) and these are mutable. Is it guaranteed by any means?
I thought the best practice was to use only the return value of the mutating call. But sure, I can change this if You wish.